lib/fsm/utils.go

```diff
@@ -43,9 +43,85 @@ func CanRollback(plan *storage.OperationPlan, phaseID string) error {
 		return trace.BadParameter(
 			"phase %q has already been rolled back", phase.ID)
 	}
+
+	// TODO: Rollback of non-leaf phases is not currently supported.
+	// Rollback starts top-down, and not in reverse order.
+	if phase.HasSubphases() {
+		return trace.BadParameter(
+			"unable to rollback non-leaf phase, only leaf phases can be rolled back").AddField("phase", phase.ID)
+	}
+
+	if !dependentsRolledBack(plan, phase.ID) {
+		return trace.BadParameter(
+			"rollback subsequent phases before rolling back phase %q", phase.ID)
```

Comment:
The check short circuits on the first dependent phase that needs to be rolled back, so we would only be able to display that first phase. I think this could be more confusing than helpful if we only display that one phase.

We could extend the check for the purpose of collecting all dependent phases. I don't think it would affect performance that much, and I think better ux might be worth it.

Reply:
Yeah let's do it :+1:

Reply:
Updated in 447efa2

Example output:
```
[vagrant@node-1 installer]$ sudo ./gravity plan rollback --phase /checks
[ERROR]: phase /checks cannot be rolled back because some phases that depend on it haven't been rolled back yet. Please rollback the following phases first:

        /pre-update
        /bootstrap

you can pass --force flag to override this check and force phase /checks rollback
```